crates/hotshot/src/tasks/task_state.rs

```diff
@@ -31,6 +37,29 @@ where
     async fn create_from(handle: &SystemContextHandle<TYPES, I>) -> Self;
 }
 
+/// A helper function to create a timeout task from a given `SystemContextHandle`.
+fn timeout_task_from_handle<TYPES: NodeType, I: NodeImplementation<TYPES>>(
+    handle: &SystemContextHandle<TYPES, I>,
+) -> JoinHandle<()> {
+    // Clone the event stream that we send the timeout event to
+    let event_stream = handle.internal_event_stream.0.clone();
+    let next_view_timeout = handle.hotshot.config.next_view_timeout;
+    let start_view = handle.hotshot.start_view;
+
+    // Spawn a task that will sleep for the next view timeout and then send a timeout event
+    // if not cancelled
+    async_spawn({
+        async move {
+            async_sleep(Duration::from_millis(next_view_timeout)).await;
+            broadcast_event(
+                Arc::new(HotShotEvent::Timeout(start_view + 1)),
+                &event_stream,
+            )
+            .await;
+        }
+    })
+}
+
```

Comment:
Nit: could this be in `impl SystemContextHandle` as e.g. `startup_timeout_task`?

Reply:
Sure, I thought about putting it there but didn't know if it was relevant to that part of the code

Reply:
I don't actually have a strong opinion -- if you think it shouldn't be there, I'm fine with it being a stand-alone function!

Reply:
@rob-maron lmk if you decide to do this, I can approve afterwards.